### PR TITLE
Adding loftr config

### PR DESCRIPTION
**Pull request by ayushbaid**
Adding LOFTR correspondence generator config. 

Sample command:
```python gtsfm/runner/run_scene_optimizer_olssonloader.py --correspondence_generator_config_name loftr.yaml --max_resolution 200```

Please note that image resolution needs to be low with LOFTR. The system runs out of memory with large images.

Result:
image

### CODE REVIEW

**Comment by akshay-krishnan** [comment]
@travisdriver or I can try to run this on a larger image resolution to see if results improve

**Comment by ayushbaid** [comment]
Ran it with `--max-resolution=400` 

3d

Looks like the results are good, but the scene visualization tool is trimming off the main scene I think.

**Comment by travisdriver** [comment]
Had some issues at lower resolutions (i.e., `max_resolution=500`) that may warrant some investigation, but a max resolution of 800 gives a great result for door.

## `--max_resolution 400`

Before BA:

Screenshot from 2022-09-21 17-28-44

After BA:

Screenshot from 2022-09-21 17-29-43

[gtsfm_metrics_report-max_res-400.zip](https://github.com/borglab/gtsfm/files/9620571/gtsfm_metrics_report-max_res-400.zip)

## `--max_resolution 500`

Before BA:

Screenshot from 2022-09-21 17-22-43

After BA:

Screenshot from 2022-09-21 17-21-36

[gtsfm_metrics_report-max_res-500.zip](https://github.com/borglab/gtsfm/files/9620512/gtsfm_metrics_report-max_res-500.zip)

## `--max_resolution 800`

Before BA:

Screenshot from 2022-09-21 17-05-25

After BA:

Screenshot from 2022-09-21 16-26-36 (1)

[gtsfm_metrics_report-max_res-800.zip](https://github.com/borglab/gtsfm/files/9620384/gtsfm_metrics_report-max_res-800.zip)

**Comment by ayushbaid** [comment]
Thanks Travis for running this. I think this result is good enough to merge. 

Also, did running it at high resolution require a lot of memory?

**Comment by travisdriver** [comment]
> Thanks Travis for running this. I think this result is good enough to merge.
> 
> Also, did running it at high resolution require a lot of memory?

Wrapping the forward call of loftr with `torch.no_grad()` fixed my OOM issues. With `max_resolution=800` I think it topped out at around 10 GB memory.

**Review by travisdriver — APPROVED** [review]
LGTM

File: gtsfm/frontend/matcher/loftr.py [hunk 1]
```diff
@@ -71,8 +72,15 @@ def match(self, image_i1: Image, image_i2: Image) -> Tuple[Keypoints, Keypoints]
...
         keypoints_i2 = Keypoints(coordinates=coords_i2_by_confidence)
-
-        return keypoints_i1, keypoints_i2
+        valid_ind = np.arange(len(keypoints_i1))
+        if image_i1.mask is not None:
+            _, valid_ind_i1 = keypoints_i1.filter_by_mask(image_i1.mask)
+            valid_ind = np.intersect1d(valid_ind, valid_ind_i1)
+        if image_i2.mask is not None:
+            _, valid_ind_i2 = keypoints_i2.filter_by_mask(image_i2.mask)
+            valid_ind = np.intersect1d(valid_ind, valid_ind_i2)
```
Comment (johnwlambert) [inline gtsfm/frontend/matcher/loftr.py:81]: thanks for the updates @travisdriver. i wonder if we should put the masking at a higher level, like in the CorrespondenceGenerator, after all detections are generated? I think we only handle it in SuperPoint elsewhere for now

Comment (travisdriver) [reply gtsfm/frontend/matcher/loftr.py:81]: I agree; should I do it in this PR or a separate one?

Comment (johnwlambert) [reply gtsfm/frontend/matcher/loftr.py:81]: Separate one is fine with me, can you open an issue for it too? Thanks!

**Review by akshay-krishnan — APPROVED** [review]

**Pull request merged**